fixtures/fixture.py

```diff
@@ -49,6 +50,13 @@ def combine_details(source_details, target_details):
         target_details[name] = content_object
 
 
+class SetupError(Exception):
+    """Setup failed.
+
+    args[0] will be a details dict.
+    """
```

Comment:
I was hesitating to add noise - args[] is the base contract for exceptions. I'll have a fiddle and see what it looks like: we can always add that later.